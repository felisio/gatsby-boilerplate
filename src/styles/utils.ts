import { css } from 'styled-components';

export const lineClamp = (numberOfLines: number) =>
    css`
        @supports (-webkit-line-clamp: ${numberOfLines}) {
            display: -webkit-box;
            -webkit-line-clamp: ${numberOfLines};
            -webkit-box-orient: vertical;
            overflow: hidden;
        }
    `;
