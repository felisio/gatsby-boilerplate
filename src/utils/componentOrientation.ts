export enum ComponentOrientation {
    horizontal = 'horizontal',
    vertical = 'vertical',
}
