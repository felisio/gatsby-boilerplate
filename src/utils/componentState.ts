export enum ComponentState {
    normal = 'normal',
    selected = 'selected',
    hover = 'hover',
    disabled = 'disabled',
}
