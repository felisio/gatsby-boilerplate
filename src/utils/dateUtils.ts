export const formatDateToString = (date: Date) =>
    date ? new Intl.DateTimeFormat('en-US', { year: 'numeric', month: 'long', day: 'numeric' }).format(date) : '';

export const sanitizeDate = (date: string | Date) =>
    date && !(date instanceof Date) ? new Date(date.replace(/-/g, '/')) : undefined;
