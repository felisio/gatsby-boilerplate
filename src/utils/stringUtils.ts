export default function generateRandomStrWithLength(length: number) {
    const alphabet = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
    return Array.from({ length })
        .map(() => {
            return alphabet.charAt(Math.floor(Math.random() * alphabet.length));
        })
        .join('');
}

export function getDateFromString(str?: string) {
    if (str) {
        const a = str.split(/[^0-9]/).map(s => {
            return parseInt(s, 10);
        });
        return new Date(a[0], a[1] - 1 || 0, a[2] || 1, a[3] || 0, a[4] || 0, a[5] || 0, a[6] || 0);
    }
    return undefined;
}
