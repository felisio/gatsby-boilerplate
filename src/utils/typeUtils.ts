export type DynamicType<T> = {
    [P in keyof T]?: T[P];
};

export interface DateTimeFormatOptions {
    localeMatcher?: 'best fit' | 'lookup';
    weekday?: 'long' | 'short' | 'narrow';
    era?: 'long' | 'short' | 'narrow';
    year?: 'numeric' | '2-digit';
    month?: 'numeric' | '2-digit' | 'long' | 'short' | 'narrow';
    day?: 'numeric' | '2-digit';
    hour?: 'numeric' | '2-digit';
    minute?: 'numeric' | '2-digit';
    second?: 'numeric' | '2-digit';
    timeZoneName?: 'long' | 'short';
    formatMatcher?: 'best fit' | 'basic';
    hour12?: boolean;
}
