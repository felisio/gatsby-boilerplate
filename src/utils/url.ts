export const isExternalLink = (str: string) => str.substr(0, 8) === 'https://' || str.substr(0, 7) === 'http://';
