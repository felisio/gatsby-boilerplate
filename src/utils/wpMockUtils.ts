import { FluidObject } from 'gatsby-image';
import { Node, WpPost, WpCategory, ImageSharp, File, WpMediaItem } from '../../graphql-types';
import generateRandomStrWithLength from './stringUtils';
import mockImage from '../assets/images/default_shared_img.png';

const february52021Date = '2020-10-30T17:16:52';

function generateDeterministicDescription(index: number) {
    return `This is the description for post ${index}. `.repeat(index + 1);
}

export function getMockAvatar() {
    return {
        src: mockImage,
        srcSet: mockImage,
        aspectRatio: 1,
        sizes: '(max-width: 40px) 100vw, 40px',
    } as FluidObject;
}

export function getMockCardBanner() {
    return {
        src: mockImage,
        srcSet: mockImage,
        aspectRatio: 1,
        sizes: '(max-width: 600px) 100vw, 600px',
    } as FluidObject;
}

function generateWpTags(n: number, nameFn: (n: number) => String) {
    return Array.from({ length: n }, (_, index) => {
        return {
            id: `${index}`,
            databaseId: index,
            children: Array<Node>({
                id: '',
                children: Array<Node>(),
                internal: { owner: '', type: '', contentDigest: '' },
            }),
            internal: { owner: '', type: '', contentDigest: '' },
            name: nameFn(index),
            uri: '#',
        };
    });
}

export function generateDeterministicWpTags(n: number) {
    return generateWpTags(n, index => {
        return `WpTag00${index}`;
    });
}

export function generateRandomWpTags(n: number) {
    return generateWpTags(n, () => {
        return generateRandomStrWithLength(Math.random() * 10 + 5);
    });
}

export function generateDeterministicWpCategories(n: number) {
    return Array.from({ length: n }, (_, index) => {
        return {
            id: `${index}`,
            databaseId: index,
            children: Array<Node>({
                id: '',
                children: Array<Node>(),
                internal: { owner: '', type: '', contentDigest: '' },
            }),
            internal: { owner: '', type: '', contentDigest: '' },
            name: `Category #${index}`,
            description: `This is the description for the category #${index}. `.repeat(index + 1),
            uri: '#',
        };
    }) as WpCategory[];
}

export function generateRandomWpPosts(n: number): Array<WpPost> {
    const imageSharp = {
        fluid: {
            src: mockImage,
            srcSet: mockImage,
            aspectRatio: 1,
            sizes: '(max-width: 600px) 100vw, 600px',
            presentationWidth: 400,
            presentationHeight: 400,
        },
    } as ImageSharp;
    const localFile = {
        childImageSharp: {
            ...imageSharp,
        },
    } as File;
    return Array.from({ length: n }).map((_, index) => {
        const newPost: WpPost = {
            id: `${index}`,
            databaseId: index,
            isSticky: false,
            categories: {
                nodes: generateDeterministicWpCategories(1),
            },
            title: `This is the blog post #${index}`,
            excerpt: `<p>${generateDeterministicDescription(index)}</p>`,
            content: '<p>This is the blog HTML content</p>',
            date: february52021Date,
            uri: 'https://quantic.edu/',
            children: [{ id: '0', children: [], internal: { contentDigest: '', owner: '', type: '' } }],
            internal: { contentDigest: '', owner: '', type: '' },
            custom: {
                horizontalImage: {
                    localFile: {
                        ...localFile,
                    },
                } as WpMediaItem,
                verticalImage: {
                    localFile: {
                        ...localFile,
                    },
                } as WpMediaItem,
            },
        };
        return newPost;
    });
}
