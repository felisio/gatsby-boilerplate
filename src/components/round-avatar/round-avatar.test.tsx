import renderer from 'react-test-renderer';
import React from 'react';
import RoundAvatar, { RoundAvatarProps } from './round-avatar';
import { getMockAvatar } from '../../utils/wpMockUtils';

const createTestProps = (props: RoundAvatarProps) => ({
    ...props,
});

test('should verify RoundAvatar Component props', () => {
    const avatarProps = createTestProps({ avatar: getMockAvatar(), alt: 'author-avatar' });
    const tree = renderer.create(<RoundAvatar {...avatarProps} />);

    expect(tree).toMatchSnapshot();
});
