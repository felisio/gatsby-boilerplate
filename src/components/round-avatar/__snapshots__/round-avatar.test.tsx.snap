// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should verify RoundAvatar Component props 1`] = `
.c0 .gatsby-image-wrapper {
  object-fit: cover;
  width: 3.2rem;
  height: 3.2rem;
  border-radius: 1.6rem;
}

@media (min-width:768px) {
  .c0 .gatsby-image-wrapper {
    width: 4rem;
    height: 4rem;
    border-radius: 2rem;
  }
}

<div
  className="c0"
>
  <div
    className=" gatsby-image-wrapper"
    style={
      Object {
        "maxHeight": null,
        "maxWidth": null,
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <div
      aria-hidden={true}
      style={
        Object {
          "paddingBottom": "100%",
          "width": "100%",
        }
      }
    />
    <picture>
      <source
        sizes="(max-width: 40px) 100vw, 40px"
        srcSet="test-file-stub"
      />
      <img
        alt="author-avatar"
        loading="lazy"
        onLoad={[Function]}
        sizes="(max-width: 40px) 100vw, 40px"
        src="test-file-stub"
        srcSet="test-file-stub"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transition": "opacity 500ms",
            "width": "100%",
          }
        }
      />
    </picture>
    <noscript
      dangerouslySetInnerHTML={
        Object {
          "__html": "<picture><source srcset=\\"test-file-stub\\" sizes=\\"(max-width: 40px) 100vw, 40px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 40px) 100vw, 40px\\" srcset=\\"test-file-stub\\" src=\\"test-file-stub\\" alt=\\"author-avatar\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
        }
      }
    />
  </div>
</div>
`;
